src/routes/charts.js

```diff
@@ -462,6 +463,16 @@ function register(server, options) {
 
     if (!hasRegisteredDataPlugins) {
         events.on(event.GET_CHART_ASSET, async function({ chart, filename }) {
+            const fn = path.join(
+                localChartAssetRoot,
+                getDataPath(chart.dataValues.created_at),
+                filename
+            );
+            try {
+                await accessAsync(fn, fs.constants.R_OK);
+            } catch (e) {
+                throw new CodedError(404);
+            }
             return fs.createReadStream(
                 path.join(localChartAssetRoot, getDataPath(chart.dataValues.created_at), filename)
```

Comment:
```suggestion
                filePath
```

Let's reuse the variable from above.